pkg/cmd/issue/lock/lock.go

```diff
@@ -0,0 +1,313 @@
+			cs.FailureIconWithColor(cs.Red),
+			currentType.FullName, issuePr.Number,
+			strings.ToLower(correctType.FullName), issuePr.Number,
+			correctType.Name, strings.ToLower(state), issuePr.Number)
+	}
+
+	successMsg := fmt.Sprintf("%s %s\n",
+		cs.SuccessIconWithColor(cs.Green), status(state, issuePr, opts))
+
+	switch state {
+	case Lock:
+		if !issuePr.Locked {
+			err = lockLockable(httpClient, baseRepo, issuePr, opts)
+		} else {
+			var relocked bool
+			relocked, err = relockLockable(httpClient, baseRepo, issuePr, opts)
+
+			if !relocked {
+				successMsg = fmt.Sprintf("%s #%d already locked%s.  Nothing changed.\n",
+					parent.FullName, issuePr.Number, reason(issuePr.ActiveLockReason))
+			}
+		}
+
+	case Unlock:
+		if issuePr.Locked {
+			err = unlockLockable(httpClient, baseRepo, issuePr, opts)
+		} else {
+			successMsg = fmt.Sprintf("%s #%d already unlocked.  Nothing changed.\n",
+				parent.FullName, issuePr.Number)
+		}
+	}
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprint(opts.IO.ErrOut, successMsg)
+
+	return nil
+}
+
+// lockLockable will lock an issue or pull request
+func lockLockable(httpClient *http.Client, repo ghrepo.Interface, lockable *api.Issue, opts *LockOptions) error {
+
+	var mutation struct {
+		LockLockable struct {
+			LockedRecord struct {
+				Locked bool
+			}
+		} `graphql:"lockLockable(input: $input)"`
+	}
+
+	variables := map[string]interface{}{
+		"input": githubv4.LockLockableInput{
+			LockableID: lockable.ID,
+			LockReason: reasonsMap[opts.Reason],
+		},
+	}
+
+	gql := graphql.NewClient(ghinstance.GraphQLEndpoint(repo.RepoHost()), httpClient)
+
+	return gql.MutateNamed(context.Background(), "LockLockable", &mutation, variables)
+}
+
+// unlockLockable will lock an issue or pull request
+func unlockLockable(httpClient *http.Client, repo ghrepo.Interface, lockable *api.Issue, opts *LockOptions) error {
+
+	var mutation struct {
+		UnlockLockable struct {
+			UnlockedRecord struct {
+				Locked bool
+			}
+		} `graphql:"unlockLockable(input: $input)"`
+	}
+
+	variables := map[string]interface{}{
+		"input": githubv4.UnlockLockableInput{
+			LockableID: lockable.ID,
+		},
+	}
+
+	gql := graphql.NewClient(ghinstance.GraphQLEndpoint(repo.RepoHost()), httpClient)
+
+	return gql.MutateNamed(context.Background(), "UnlockLockable", &mutation, variables)
+}
+
+// relockLockable will unlock then lock an issue or pull request.  A common use
+// case would be to change the reason for locking.
+func relockLockable(httpClient *http.Client, repo ghrepo.Interface, lockable *api.Issue, opts *LockOptions) (bool, error) {
+
+	var relocked bool
+	shouldRelock := &survey.Confirm{
+		Message: fmt.Sprintf("%s #%d already locked%s.  Unlock and lock again%s?",
+			alias[opts.ParentCmd].FullName, lockable.Number, reason(lockable.ActiveLockReason), reason(opts.Reason)),
+		Default: true,
+	}
+
+	err := survey.AskOne(shouldRelock, &relocked)
```

Comment:
This should:

- use the new `Prompter`
- not prompt if STDIN is not a tty (see `IOstreams.CanPrompt` )